### DISCUSSION TITLE
Is it possible to batch updates in multiple useSwr calls with unstable_batchedUpdates?

### DISCUSSION THREAD

**Discussion by Linksku** [vercel/swr#1146] [Q&A]
I have a batched API, all the useSwr calls on the same tick are sent to the server in one request. When the request completes, it looks like useSwr calls setState multiple times without unstable_batchedUpdates, so it causes multiple rerenders. Would it be possible to only rerender once when multiple fetchers complete at the same time?

**Comment by isBatak** [comment]
You can have one fetcher with multiple API calls.
```
const { data } = useSWR(() => 'some-unique-key', () => Promise.all([fetch1, fetch2, fetch3]));
```
But this seems like a bad Rest API design. You should be able to fetch one endpoint and get a compound document with all three resources.

**Comment by Linksku** [comment]
It's not a REST API, it looks a bit like:

```
const queuedKeys = {};
let timer;
useSWR('key', key => {
  if (!timer) {
    timer = setTimeout(handleQueuedKeys, 0);
  }
  return new Promise(succ => {
    queuedKeys[key] = succ;
  });
});
```
  
`handleQueuedKeys` would batch the APIs and send them to the server. When the request completes, multiple promises resolve and `useSWR` calls `setState` multiple times. I want to batch these `setState`s.

**Reply by shuding** [reply]
I don’t think this is possible. How can multiple custom hooks batch their internal setState?
